components/script/dom/htmlimageelement.rs

```diff
@@ -433,15 +434,21 @@ impl HTMLImageElement {
         };
 
         let map = self.upcast::<Node>()
-                      .following_siblings()
+                      .GetRootNode()
+                      .traverse_preorder()
                       .filter_map(Root::downcast::<HTMLMapElement>)
                       .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
 
-        let elements: Vec<Root<HTMLAreaElement>> = map.unwrap().upcast::<Node>()
-                      .children()
+        match map {
+            None => None,
+            Some(mapElem) => {
+                let elements: Vec<Root<HTMLAreaElement>> = mapElem.upcast::<Node>()
+                      .traverse_preorder()
                       .filter_map(Root::downcast::<HTMLAreaElement>)
                       .collect();
-        Some(elements)
+                Some(elements)
+            }
+        }
```

Comment:
This would be shorter as `map.map(|elem| { ... })`, probably with a different name for `map`, for clarity.